rfc/rfc-79/rfc-79-2.md

```diff
@@ -0,0 +1,99 @@
+
+In order to ensure that other writers can indeed permenantely cancel a cancellable table service plan (such that it can no longer be executed), additional changes to clean and table service flow will be need to be added as well. Two proposals are detailed below. Also, note that the cancellation-policy is only required to be honored by clean: a user can choose setup an application to aggresively clean up a failed cancellable table service plan even if it has not meet the critera for its cancellation-policy yet. This can be useful if a user wants a utility to manually ensure that clean/archival for a dataset progresses immdeitately or knows that a cancellable table service plan will not be attempted again or cleaned up by another writer. Each proposal provides an example on how to achieve this.
+
+#### (A) Making cancellable plans "mutable" 
+Cancellable table service plans can be updated to have a "mutuable" plan, in the sense that once a plan is transitioned to inflight, if the execution of it fails the plan must be rolled back and deleted, similar to rollback of failed ingestion writes. The flow for table service execution will be similar to the existing one for immutable plan, except that if the plan is targeted by a rollback plan its execution will abort.
+
+![mutable table service lifecycle (3)](https://github.com/user-attachments/assets/7c7d67f5-b6d3-44e6-990b-340920bc16a7)
+
+Once cancellable table service plans are made mutable in this manner, clean can rollback failed cancellable table service plans that have met the cancellation-policy critera, similar to how clean currently rolls back failed ingestion writes. Specifically, clean can check for any failed cancelled table service plans that are already part of a pending rollback plan or meet the cancellation-policy. From there a rollback can be scheduled/executed for each instant.
+With these changes, a failed cancellable table service plan that has met its cancellation policy will be guaranteed to be attempted for rollback by the next clean. If a user wants to immeditaly cleanup a failed cancellable plan, they can bypass the cancellation policy by scheduling and executing a rollback plan, the same way that clean will cleanup these plans.
+
+This meets the critera for goal (2). But comes with the following drawback:
+* The instant metadata file for the cancellable table service plan will be deleted on rollback, analogous to how rollback of a ingestion instant works. This can make it more difficult to debug failed/stuck cancellable table service plans
+  
+#### (B) Adding a cancel operation/state for cancellable plans
+An alternate approach can involve updating the possible tmeline actions / states, by making the following changes:
```

Comment:
in this approach, when exactly the table service plan.requested meta file in timeline will be deleted ?

Reply:
Oh I mentioned under drawbacks that archival will now have to be updated to handle deleting those. 

 > Readers need to account for skipping aborted plans, and archival should clean up aborted plans and cancel actions accordingly. In addition, it is now possible for an instant to be transitioned to a terminal state (.aborted) without being completed. This can complicate tooling that needs to infer wether an instant is comitted.
 
 since now an instant can be in a terminal state while still being completed. But I can mention this in design part.